### PR TITLE
use local config for vault when masterless

### LINKED ISSUE #45976: 6a5e0f9 introduces regression that breaks Vault module for salt masterless
### Description of Issue/Question

https://github.com/saltstack/salt/commit/6a5e0f9ac11e7731b759d7dcc873c07f6436ba1b introduces a regression that breaks the Vault module for salt masterless.

Since https://github.com/saltstack/salt/commit/6a5e0f9ac11e7731b759d7dcc873c07f6436ba1b, we get the following error message running `salt-call --local state.apply`:

        [CRITICAL] Rendering SLS 'foo' failed, render error:
        Problem running salt function in Jinja template: [Errno 2] No such file or directory: '/etc/salt/pki/minion/minion.pem'; line 9

        ---
        [...]
            foo:
              - "{{ salt.sdb.get('sdb://my-vault/secret/foo?value') }}"    <======================
        ---

### Setup

We use [Packer](https://packer.io) to generate machine images, with Salt masterless for configuration management. This change prevents us from being able to build new images because it breaks the Vault module for Salt masterless.

### PR DESCRIPTION
**Pull request by gtmanfred**
### What does this PR do?
Dont contact master when running a masterless minion (this includes salt-ssh)

### What issues does this PR fix or reference?
Fixes #45976

### Tests written?

No

### Commits signed with GPG?

Yes

### CODE REVIEW

File: salt/utils/vault.py [hunk 1]
```diff
@@ -111,8 +111,8 @@ def _use_local_config():
 
     if 'vault' in __opts__ and __opts__.get('__role', 'minion') == 'master':
         return _use_local_config()
-    elif '_ssh_version' in __opts__:
-        return _use_local_config()
+    elif any((__opts__['local'], __opts__['file_client'] == local, __opts__['master_type'] == 'disable')):
+        _use_local_config()
```
Comment (lorengordon) [inline salt/utils/vault.py]: Did you mean to delete the `return`?

Comment (gtmanfred) [reply salt/utils/vault.py]: yes it does, and i have no idea where it went, cause i only changed the _ssh_version line

**Review by garethgreenaway — APPROVED** [review]

**Comment by lorengordon** [comment]
Should this base on 2017.7.3 (as it is), or just 2017.7?

**Comment by gtmanfred** [comment]
This is the correct branch

**Review by terminalmage — APPROVED** [review]

**Pull request merged**